src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1647,6 +1673,19 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                     _pool.Free(constraints);
                 }
             }
+
+            SyntaxToken? eatRecordModifierIfAvailable()
+            {
+                Debug.Assert(keyword.Kind == SyntaxKind.RecordKeyword);
+                if (CurrentToken.Kind is SyntaxKind.ClassKeyword or SyntaxKind.StructKeyword)
+                {
+                    var result = EatToken();
+                    result = CheckFeatureAvailability(result, MessageID.IDS_FeatureRecordStructs);
+                    return result;
+                }
```

Comment:
For `record class`, this will generate an error that `record struct` feature isn't available.
While generating an error is reasonable, the error message feels weird.

Reply:
Yeah. Considered the problem, but I don't think it's worth it to add a "record class" feature. Would seem even more confusing.